packages/main/src/plugin/container-registry.ts

```diff
@@ -770,6 +770,32 @@ export class ContainerProviderRegistry {
     }
   }
 
+  async addImageTag(
```

Comment:
since you created the PR there is now https://github.com/containers/podman-desktop/blob/a9cb79c302c47c3589fdc208e145baeb77be2113/packages/main/src/plugin/container-registry.ts#L680 so I think you can get rid of this method and use the existing one

Reply:
I still need a callback so I can change the state of the button in the model. Should I just call `tagImage` from `addImageTag` in the container-registry file? I will also need to remove tags if the (new) checkbox is checked which I can do in `addImageTag`.

Reply:
Hello @tuckerrc I don't think you need a callback.

Callback is usually used when you need data before the end of the method (like streaming of the output)
But here, you only need to know when it has been done

so, when you call the method, just wait it has finished

instead of doing
```
async function renameImage(imageTag: string, newImageName: string, newImageTag: string) {
  await window.addImageTag(imageInfoToRename.engineId, imageTag, newImageName, newImageTag, callback);
}
```

you can do
```
async function renameImage(imageTag: string, newImageName: string, newImageTag: string) {
  await window.addImageTag(imageInfoToRename.engineId, imageTag, newImageName, newImageTag);
  renameComplete = true;
}
```
and get rid of callback

Reply:
I ended up removing the callback and could just call the image tag methods directly